pkg/nuctl/command/common/patchmanifest.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/nuclio/logger"
+)
+
+type PatchManifest struct {
+	*patchManifest
+	lock sync.Mutex
+}
+
+type patchManifest struct {
+	Success []string                   `json:"success,omitempty"`
+	Skipped []string                   `json:"skipped,omitempty"`
+	Failed  map[string]FailDescription `json:"failed,omitempty"`
+}
+
+type FailDescription struct {
+	Err       string `json:"error,omitempty"`
+	Retryable bool   `json:"retryable"`
+}
+
+func NewPatchManifest() *PatchManifest {
+	return &PatchManifest{
+		lock: sync.Mutex{},
+		patchManifest: &patchManifest{
+			Success: []string{},
+			Skipped: []string{},
+			Failed:  make(map[string]FailDescription),
+		},
+	}
+}
+
+func NewPatchManifestFromFile(path string) *PatchManifest {
+	parsedManifest, err := readManifestFromFile(path)
+	if err != nil {
+		return nil
```

Comment:
Why are you ignoring the error?
You can return it as well and handle it from the outside as well.